crud/select.lua

```diff
@@ -137,32 +142,33 @@ local function build_select_iterator(space_name, user_conditions, opts)
     if space == nil then
         return nil, SelectError:new("Space %s doesn't exists", space_name)
     end
-
     local space_format = space:format()
 
-    local after_tuple = utils.flatten(opts.after, space_format)
-
     -- plan select
-    local plan, err = select_plan.new(space, conditions, {
-        limit = opts.limit,
-        after_tuple = after_tuple
-    })
+    local plan, err = select_plan.gen_by_conditions(space, conditions)
 
     if err ~= nil then
         return nil, SelectError:new("Failed to plan select: %s", err)
     end
 
-    -- get replicasets to select from
-    local replicasets, err = get_replicasets_to_select_from(plan, replicasets)
-    if err ~= nil then
-        return nil, SelectError:new("Failed to get replicasets to select from: %s", err)
-    end
+    -- set limit and replicasets to select from
+    local limit = opts.limit
+    local replicasets_to_select = replicasets
 
-    local scan_index = space.index[plan.scanner.index_id]
+    local scan_index = space.index[plan.index_id]
     local primary_index = space.index[0]
+    if select_plan.is_scan_by_full_sharding_key_eq(plan, scan_index, primary_index) then
+        limit = 1
+        plan.iter = box.index.REQ
+
+        replicasets_to_select = get_replicaset_by_scan_value(plan.scan_value)
+    end
+
+    -- set after tuple
+    local after_tuple = utils.flatten(opts.after, space_format)
```

Comment:
Not sure it's a good place for unflatten...

Reply:
I believe we should pass raw tuple to this function as option...

But ok, feel free to ignore this comment

Reply:
Since crud returns map, user will use it as an `after` value for selecting next batch of data.